### PR TITLE
Box: Add docs and story for Box component

### LINKED ISSUE #509: Add documentation for Box component

### PR DESCRIPTION
**Pull request by LucianBuzzo**
Fixes #509

Change-type: patch
Signed-off-by: Lucian Buzzo <lucian.buzzo@gmail.com>

### CODE REVIEW

**Comment by thundron** [comment]
Looks good code-wise, but shouldn't the commit text follow the guidelines from [COMMIT-GUIDELINES](https://github.com/resin-io/etcher/blob/master/docs/COMMIT-GUIDELINES.md) and [Commit_and_PR_Guidelines](https://github.com/resin-io/process/blob/master/process/Commit_and_PR_Guidelines.md#commit-guidelines)?

**Comment by LucianBuzzo** [comment]
@thundron Could you specific about what should be changed in the commit text?

**Comment by thgreasi** [comment]
@thundron let me try to explain this urban knowledge:
* `Fixes #509` instead of `Connects-To: #509`
  * Since we are not using waffle in this repo, we are going using the GH syntax instead so that the issues get closed once the PR is merged
* `Change-type` instead of `Change-Type`
  * We were indeed following the [Commit_and_PR_Guidelines](https://github.com/resin-io/process/blob/master/process/Commit_and_PR_Guidelines.md#commit-guidelines) but since we started using [resin-commit-lint](https://github.com/resin-io/resin-commit-lint) in all repos, we are following that one instead. I will open an issue in the process repo to update the PR guidelines.

**Comment by thgreasi** [comment]
@thundron 
See: https://github.com/resin-io/process/pull/188

**Comment by thundron** [comment]
Oh but I meant to ask a *much* simpler question: shouldn't the commit be something along the lines of `feat|fix(DOC): Add Box story and docs` instead of `Box: Add docs and story for Box component` ?

**Comment by thgreasi** [comment]
Oh I see @thundron , I wasn't aware of the etcher commit conventions. They seem great and they remind me the ones used by the angular community.
I do like something like `docs(Box): document and add story for the Box component` but I wasn't in any discussion for how the commit prefix should look in general. @nazrhom ?

I will leave it up to Lucian to decide for this one and we should have a discussion on this at some point.

File: src/stories/README/Box.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+The basic building block of a rendition application.
+
+[View story source](https://github.com/resin-io-modules/rendition/blob/master/src/stories/Box.js)
+
+## Props
+
+| Name          | Type      | Default   | Required   | Description                                          |
+| --------------------------------------------------------------------------------------------------------- |
+| `flex`      | <code>string &#124; string[]</code> | - | - | Sets `flex`, if the value is an array, sets a responsive style corresponding to the theme's breakpoints
```
Comment (thgreasi) [inline src/stories/README/Box.md]: I think that we also have `order` in the typings.
It feels very poor that we do not mention at all any of the DefaultProps :thinking: even though the story does demonstrate some of them.

**Comment by thundron** [comment]
It's just an idea to make it *look* more clean, as I said it's nothing life-changing so up to you guys, for everything else looks good to me.

File: src/stories/README/Box.md (hunk 1 above)
Comment (LucianBuzzo) [reply src/stories/README/Box.md]: @thgreasi In the main README we mention that **all** components utilise the styled system props. I'm open to suggestions on how we can make that clearer

**Pull request merged**